### PR TITLE
Feature/add documentation

### PR DESCRIPTION
**Pull request by tinytyranid**
This PR adds:
- Documentation with several examples
- A fix to a bug where not all lines were duplicated properly
    - Because not all columns were used for sorting before duplication sometimes some lines would escape deduplication. I have therefore switched to deduplicating using a map.

### CODE REVIEW

File: docs/sorting.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+GL000209.1	10	11	1	A
+1	10	11	-1	A
+1	12	13	1	A
+X	10	11	1	A
+1	10	11	1	A
+1	10	11	-1	B
+MT	10	11	1	A
+```
+
+## Lexicographic Sorting
```
Comment (marrip) [inline docs/sorting.md:23]:
```suggestion
## Lexicographic Sorting (default)
```

File: docs/sorting.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+Lexicographic sorting is the default sorting method.
```
Comment (marrip) [inline docs/sorting.md:25]:
```suggestion
```

File: docs/sorting.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+10      12      13      1       D
+X       10      11      1       A
+Y       10      11      1       A
+MT      10      11      1       A
+GL000209.1      10      11      1       A
+```
+
+### Custom Chromosome Order
+
+Adding the option custom chromosome sorting one can sort the chromosomes in any order that one would want with the flag `--chr-order`. Chromosomes not on the list will be sorted naturally after the ones in the list.
```
Comment (marrip) [inline docs/sorting.md:152]:
```suggestion
Adding the option custom chromosome sorting, one can sort the chromosomes in any order that one would want with the flag `--chr-order`. Chromosomes not on the list will be sorted naturally after the ones in the list.
```

**Review by marrip — APPROVED** [review]
LGTM! I have some minor suggestions. Well done, Camilla! 👏

File: README.md [hunk 4]
```diff
@@ -2,9 +2,104 @@
...
+|                                     |                         |                                                                                                                                                                                                                                                                   |
+| **merging**                         |                         |                                                                                                                                                                                                                                                                   |
+| `--no-merge`                        | `NO_MERGE`              | Do not merge bed regions. Note that touching regions are merged (e.g. if two regions are on the same chr they will be merged if one ends at 5 and the other starts at 6)                                                                                          |
```
[the remark this answers is not in the document]
Comment (tinytyranid) [reply README.md:104]: This seems to be a remnant of when merging was optional. I will move this information over to the information for `overlap`.

File: README.md [hunk 5]
```diff
@@ -2,9 +2,104 @@
...
+BedFusion follows the bed file standard outlined in: [Niu J., Denisko D. & Hoffman M. M. (2022): *The Browser Extensible Data (BED)* format](https://github.com/samtools/hts-specs/blob/94500cf76f049e898dec7af23097d877fde5894e/BEDv1.pdf)
+
+## Quick-Start Guide
+
+BedFusion will both sort (lexicographically) and merge regions by default. 
+
+Example bed file `examples/merge-test.bed`:
+
+``` text
+1	1	4	1	X
```
[the remark this answers is not in the document]
Comment (tinytyranid) [reply README.md:16]: Good point, I will change these A and B, like in the sort example.

File: docs/sorting.md (hunk 2 above)
Comment (tinytyranid) [reply docs/sorting.md:25]: I think I want to double down on this information, so I'm keeping this sentence even if we are also adding it to the title :)

Comment (marrip) [reply docs/sorting.md:25]: hm, then I would rather not add it to the title. I would understand if you repeated it in a different section but having the same information twice in two consecutive lines seems a bit much 😬

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tinytyranid) [reply README.md]: I have now updated this information in d757b77c1c652660ac37d610bec0909fbf973dd4

Please tell me what you think :)

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (tinytyranid) [reply README.md]: Fixed in 812ff891b4f31c9a853a59b23f003f065e63fa2c

File: README.md [hunk 6]
```diff
@@ -2,9 +2,103 @@
...
+| **merging**                         |                         |                                                                                                                                                                                                                                                                             |
+| `--no-merge`                        | `NO_MERGE`              | Do not merge regions                                                                                                                                                                                                                                                        |
+| `--overlap=0`                       | `OVERLAP`               | Overlap between regions to be merged. Note that touching regions are merged (e.g. if two regions are on the same chr, and the overlap is they will be merged if one ends at 5 and the other starts at 6). If you don't want touching regions to be merged set overlap to -1 |
```
Comment (marrip) [inline README.md:104]:
```suggestion
| `--overlap=0`                       | `OVERLAP`               | Overlap between regions to be merged. Note that touching regions are merged (e.g. if two regions are on the same chr, and the overlap is `0` they will be merged if one ends at 5 and the other starts at 6). If you don't want touching regions to be merged set overlap to `-1` |
```

**Pull request merged**